File: exercises/regex_book/regex.py
import re

def p(regex, text):
    print(re.findall(regex,
                     text,
                     flags=re.MULTILINE |
                     re.IGNORECASE))

# text = ("cat\ncot\nCATASTROPHE\nWILDCAUGHT\n" +
#         "wildcat\n-GET-\nYacht")

# p(r'^c.t', text) # ['cat', 'cot', 'CAT']
# p(r'c.t$', text) # ['cat', 'cot', 'cat', 'cht']

p(r'\Afoo\Z', 'foo') # ['foo']
p(r'\Afoo\Z', 'fool') # []
p(r'\Afoo\Z', 'barfoo') # []

File: exercises/regex_book/regex.js
function p(str, regex) {
    console.log(str.match(regex));
  }
  
  let text = "cat\ncot\nCATASTROPHE\nWILDCAUGHT\n" +
             "wildcat\n-GET-\nYacht"
  
  p(text, /^c.t/mig) // [ 'cat', 'cot', 'CAT' ]
  p(text, /c.t$/mig) // [ 'cat', 'cot', 'cat', 'cht' ]